File: DailyLeetcode/13Aug/PowerOfThree.java
//Solved by Iterative Division
//This solution checks if a number is a power of three by continuously dividing it by 3
//Time Complexity: O(log n)
//Space Complexity: O(1)
class Solution {
    public boolean isPowerOfThree(int n) {
        if(n<=0) return false;
        while(n%3==0){
            n/=3;
        }
        return n==1;
    }
}

File: DailyLeetcode/13Aug/PowerOfThree.cpp
//Recursion
//Time Complexity:O(logn)
//Space Complexity:O(1)
class Solution {
public:
    bool isPowerOfThree(int n) {
        if(n==0 || n==1){
            return  n;
        }
        if(n%3!=0){
            return 0;
        }
        return isPowerOfThree(n/3);
    }
};